core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -328,7 +328,7 @@ object SparkSubmit extends CommandLineUtils with Logging {
         printErrorAndExit("Cluster deploy mode is not applicable to Spark shells.")
       case (_, CLUSTER) if isSqlShell(args.mainClass) =>
         printErrorAndExit("Cluster deploy mode is not applicable to Spark SQL shell.")
-      case (_, CLUSTER) if isThriftServer(args.mainClass) =>
+      case (_, CLUSTER) if (clusterManager != KUBERNETES) && isThriftServer(args.mainClass) =>
         printErrorAndExit("Cluster deploy mode is not applicable to Spark Thrift server.")
```

Comment:
I haven't dug through but this might break if there has been any assumption that thrift is not running in cluster mode?

Reply:
Can you elaborate on what might go wrong? I have been using it successfully to run queries, is there anything else that should be tested?